src/components/AddItemForm.js

```diff
@@ -1,28 +1,64 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import fb from '../lib/firebase';
 import '../css/AddItemForm.css';
-import { v4 as uuidv4 } from 'uuid';
 
 const Form = ({token}) => {
     const [itemName, setItemName] = useState("");
     const [timeFrame, setTimeFrame] = useState(7);
     const [lastPurchaseDate, setPurchaseDate] = useState(null);
     const userToken = token || "userToken";
+    const [shoppingListCollection, setShoppingListCollection] = useState([]);
+    const [duplicateError, setDuplicateError] = useState(false);
+
+    useEffect(() => {
+        const db =  fb.firestore()
+        const tokenRef = db.collection(token);
+
+        tokenRef
+            .orderBy("timeFrame", "asc")
+            .get()
+            .then((querySnapshot) => {
+                let fullCollection = [];
+                if(!querySnapshot.empty){
+                    querySnapshot.forEach((doc) => {
+                        let documentData = doc.data();
+                        let nameData = documentData.itemName
+                        if(!nameData){
```

Comment:
This should be 
```suggestion
                        if(nameData){
```
you want it to run this if there is `nameData`, right?